sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -224,6 +224,22 @@ static const struct dmi_system_id sof_rt5682_quirk_table[] = {
 					SOF_SSP_BT_OFFLOAD_PRESENT
 					),
 	},
+	{
+		.callback = sof_rt5682_quirk_cb,
+		.matches = {
+			DMI_MATCH(DMI_PRODUCT_FAMILY, "Google_Rex"),
+			DMI_MATCH(DMI_OEM_STRING, "AUDIO-MAX98360_ALC5682I_DISCRETE_I2S_BT"),
+		},
+		.driver_data = (void *)(SOF_RT5682_MCLK_EN |
+					SOF_RT5682_SSP_CODEC(2) |
+					SOF_SPEAKER_AMP_PRESENT |
+					SOF_MAX98360A_SPEAKER_AMP_PRESENT |
+					SOF_RT5682_SSP_AMP(0) |
+					SOF_RT5682_NUM_HDMIDEV(4) |
```

Comment:
@plbossart Do we want to support 4 HDMI on MTL I2S machines?

Reply:
no, we added 3 only in topology2 files.

Reply:
> no, we added 3 only in topology2 files.

@udaymb, then  SOF_RT5682_NUM_HDMIDEV should be 3, and same as line 222.

Reply:
Can we add a quirk  SOF_PDM_NUM_CHANNELS() and load the different topologies accordingly based on the DMI information?

Reply:
This quirk is not necessary since SOF_SPEAKER_AMP_PRESENT and SOF_*_SPEAKER_AMP_PRESENT are removed.

Reply:
@bardliao - modified SOF_RT5682_NUM_HDMIDEV to 3
@brentlu - rebased it to get the latest changes and based on that removed these quirk entries SOF_SPEAKER_AMP_PRESENT and SOF_*_SPEAKER_AMP_PRESENT

Reply:
Updated topology - https://github.com/thesofproject/sof/pull/8131